### PR TITLE
Added abilty to customize some components shape

### PR DESCRIPTION
**Pull request by hubertmalkowski**
1. I've added css shape tokens from official docs [material design](https://m3.material.io/styles/shape/shape-scale-tokens#a830b8f8-f15b-4be9-8b2e-839f78481cf2)
2. Components with shape customization variable: 
    - Button
    - ButtonLink
    - FAB
    - SegmentedButton
    - BottomSheet
    - Card
    - CardClickable
    - Dialog
    - Menu
    - Chip
    - TextField (each type)
    - DateField

### CODE REVIEW

**Review by KTibow — CHANGES_REQUESTED** [review]
- :root is global without :global
- review the changes you made and undo ones that are out of scope

**Comment by KTibow** [comment]
do you have plans to finish doing this for the rest of components or should i implement them

**Comment by KTibow** [comment]
the snackbar is... complicated. it needs to accomplish the following things:
- allow the box-shadow to not get clipped
- approximate a border radius with 2 intermediate points
- be able to animate from one form to another

**Comment by hubertmalkowski** [comment]
I guess that's all except snackbar

File: src/lib/forms/SliderTicks.svelte [hunk 1]
```diff
@@ -88,7 +92,7 @@
   }
   .track {
     width: 100%;
-    height: 0.25rem;
+    height: var(--m3-slider-track-shape);
```
Comment (KTibow) [inline src/lib/forms/SliderTicks.svelte:95]: surely you meant to change the border radius not height

**Comment by KTibow** [comment]
merging this but not releasing a new version. i'd like it if people tried it out on the website before this breaks anything.

**Pull request merged**